src/test.js

```diff
@@ -737,23 +738,41 @@ function internalStart( test ) {
 	}
 }
 
-function numberOfTests( module ) {
-	let count = module.tests.length;
+function collectTests( module ) {
+	const tests = [].concat( module.tests );
 	const modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
 	while ( modules.length ) {
 		const nextModule =  modules.shift();
-		count += nextModule.tests.length;
+		tests.push.apply( tests, nextModule.tests );
 		modules.push( ...nextModule.childModules );
 	}
 
-	return count;
+	return tests;
+}
+
+function numberOfTests( module ) {
+	return collectTests( module ).length;
+}
+
+function numberOfUnskippedTests( module ) {
+	return collectTests( module ).filter( test => !test.skip ).length;
 }
 
-function notifyTestsRan( module ) {
+function notifyTestsRan( module, skipped ) {
+	const thisModule = module;
+
 	module.testsRun++;
 	while ( ( module = module.parentModule ) ) {
 		module.testsRun++;
 	}
+
+	if ( !skipped ) {
```

Comment:
I cringe at checking a loop invariant inside the loop, although this time I guess I went the direction of incurring a performance cost in the common case rather that the uncommon one :man_facepalming: 

How about this:

```
let notify;
if ( skipped ) {
  notify = function(module) {
    module.testsRun++;
  };
} else {
  notify = function(module) {
    module.testsRun++;
    module.unskippedTestsRun++;
  };
}

notify(module);
while ( ( module = module.parentModule ) ) {
  notify(module);
}
```

Not a big deal if you don't like it and prefer just checking the invariant inside the loop -- let me know.